### ISSUE TITLE
recognize still does not seem to respect the configured CPU limit, overloading the server (especially RAM in case of smaller memory size)

### ISSUE THREAD

**Issue by gohrner** [nextcloud/recognize#729]
### Which version of recognize are you using?

3.6.2

### Enabled Modes

Object recognition, Face recognition, Video recognition, Music recognition

### TensorFlow mode

Normal mode

### Which Nextcloud version do you have installed?

25.0.4

### Which Operating system do you have installed?

Debian 11 "Bullseye"

### Which Docker container are you using to run Nextcloud? (if applicable)

--

### How much RAM does your server have?

4 GiB

### What processor Architecture does your CPU have?

x86_64

### Describe the Bug

Probably similar to #335, but as this one is closed, it does not seem to get attention any more, despite multiple comments ([1](https://github.com/nextcloud/recognize/issues/335#issuecomment-1442413007), [2](https://github.com/nextcloud/recognize/issues/335#issuecomment-1464878349), [3](https://github.com/nextcloud/recognize/issues/335#issuecomment-1465024692)) about the CPU limiter not working.

Yesterday and today, my server was basically rendered unusable due to recognize starting (at least) 9 `node` instances in parallel, totally overloading the system's memory.

A maximum of "1 CPU" was / is configured.

I now had to disable recognize completely.

### Expected Behavior

At most as many classification processes as configured by the CPU setting should be started.

### To Reproduce

Enable recognize (for all supported kinds of classifications), set "1 CPU" as the limit, and count the `node` processes.

**Comment by marcelklehr** [comment]
Thank you for opening a new issue to get this to my attention!

> Yesterday and today, my server was basically rendered unusable due to recognize starting (at least) 9 node instances in parallel, totally overloading the system's memory.

Are these node processes started by different cron.php processes?

**Comment by gohrner** [comment]
> Are these node processes started by different cron.php processes?

I would need to check using `pstree`, but have to reproduce this issue to do so - but I have to do this during times where the server is not really needed, as it will impact service. It's only a private machine, but performs email delivery for the family... ;)

**Comment by gohrner** [comment]
I tried to do the test run last night, but despite enabling recognize in the late afternoon yesterday, it does not seem to have run any classification during the night and no node processes were present at all this morning.

I have to disable recognize again for the day now and can retry this evening. Any hints on how to force recognize to actually start classification once it's enabled in the admin frontend?

**Comment by gohrner** [comment]
Same tonight - I enabled recognize yesterday evening, but no running node processes today at all, and more than 9000 unclassified images according to the admin frontend. I now disabled it again for the day.

I'd be glad about any information on how to actually trigger classification runs so I can test this. Otherwise I'll try to keep it enabled for somewhat longer during the weekend.

**Comment by marcelklehr** [comment]
Could you post the contents of the oc_jobs table?

**Comment by gohrner** [comment]
@marcelklehr: I didn't explicitly observe multiple running node processes since then, which is why I didn't reply back so far. (I was waiting for it to happen again.)

Just now, the server again didn't respond, but when I managed to connect after some waiting time, there also was only one node process running - however maybe another one just stopped, I don't know.

Here's the current jobs table:

```
MariaDB [nextcloud_db]> SELECT * FROM oc_jobs;
+-------+-----------------------------------------------------------------+----------------------------+------------+--------------+-------------+--------------------+----------------------------------+----------------+
| id    | class                                                           | argument                   | last_run   | last_checked | reserved_at | execution_duration | argument_hash                    | time_sensitive |
+-------+-----------------------------------------------------------------+----------------------------+------------+--------------+-------------+--------------------+----------------------------------+----------------+
|     1 | OCA\Files_Versions\BackgroundJob\ExpireVersions                 | null                       | 1680248402 |   1680250124 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     2 | OCA\Activity\BackgroundJob\EmailNotification                    | null                       | 1680250120 |   1680250120 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     3 | OCA\Activity\BackgroundJob\ExpireActivities                     | null                       | 1680198002 |   1680250120 |           0 |                 26 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|     4 | OCA\Files_Sharing\DeleteOrphanedSharesJob                       | null                       | 1680249602 |   1680250120 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     5 | OCA\Files_Sharing\ExpireSharesJob                               | null                       | 1680201601 |   1680250120 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|     6 | OCA\Files_Sharing\BackgroundJob\FederatedSharesDiscoverJob      | null                       | 1680183002 |   1680250120 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     7 | OCA\Support\BackgroundJobs\CheckSubscription                    | null                       | 1680250121 |   1680250121 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     8 | OCA\UpdateNotification\Notification\BackgroundJob               | null                       | 1680183002 |   1680250121 |           0 |                 60 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     9 | OCA\Federation\SyncJob                                          | null                       | 1680207001 |   1680250121 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    10 | OCA\NextcloudAnnouncements\Cron\Crawler                         | null                       | 1680210902 |   1680250121 |           0 |                  6 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    12 | OCA\DAV\BackgroundJob\CleanupDirectLinksJob                     | null                       | 1680201602 |   1680250121 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|    13 | OCA\DAV\BackgroundJob\UpdateCalendarResourcesRoomsBackgroundJob | null                       | 1680248701 |   1680250121 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    14 | OCA\DAV\BackgroundJob\CleanupInvitationTokenJob                 | null                       | 1680198028 |   1680250121 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|    15 | OCA\Files\BackgroundJob\ScanFiles                               | null                       | 1680249602 |   1680250121 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    16 | OCA\Files\BackgroundJob\DeleteOrphanedItems                     | null                       | 1680248701 |   1680250121 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    17 | OCA\Files\BackgroundJob\CleanupFileLocks                        | null                       | 1680250122 |   1680250121 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    18 | OCA\Files_Trashbin\BackgroundJob\ExpireTrash                    | null                       | 1680250122 |   1680250122 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    19 | OC\Authentication\Token\DefaultTokenCleanupJob                  | null                       | 1665093689 |   1680262202 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    20 | OC\Log\Rotate                                                   | null                       | 1680250122 |   1680250122 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    21 | OC\Preview\BackgroundCleanupJob                                 | null                       | 1680248701 |   1680250122 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    75 | OCA\UpdateNotification\ResetTokenBackgroundJob                  | null                       | 1680249602 |   1680250122 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    81 | OCA\Richdocuments\Backgroundjobs\ObtainCapabilities             | null                       | 1680248702 |   1680250122 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    87 | OC\Core\BackgroundJobs\CleanupLoginFlowV2                       | null                       | 1680248702 |   1680250122 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|   105 | OCA\DAV\BackgroundJob\EventReminderJob                          | null                       | 1680250122 |   1680250122 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|   106 | OCA\Text\Cron\Cleanup                                           | null                       | 1680250122 |   1680250122 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|   141 | OCA\Polls\Cron\NotificationCron                                 | null                       | 1680250123 |   1680250122 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|   234 | OCA\Files\BackgroundJob\CleanupDirectEditingTokens              | null                       | 1680250123 |   1680250123 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|   236 | OCA\WorkflowEngine\BackgroundJobs\Rotate                        | null                       | 1680246003 |   1680250123 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  1761 | OCA\ContactsInteraction\BackgroundJob\CleanupJob                | null                       | 1680201902 |   1680250123 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  3070 | OCA\Contacts\Cron\SocialUpdateRegistration                      | null                       | 1679938801 |   1680250123 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  3083 | OCA\Talk\BackgroundJob\ExpireSignalingMessage                   | null                       | 1680250123 |   1680250123 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3084 | OCA\Talk\BackgroundJob\RemoveEmptyRooms                         | null                       | 1680250123 |   1680250123 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  3085 | OCA\Talk\BackgroundJob\ResetAssignedSignalingServer             | null                       | 1680250124 |   1680250124 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3161 | OCA\Activity\BackgroundJob\DigestMail                           | null                       | 1680248702 |   1680250124 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3163 | OCA\UserStatus\BackgroundJob\ClearOldStatusesBackgroundJob      | null                       | 1680250124 |   1680250124 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3182 | OCA\Talk\BackgroundJob\CheckHostedSignalingServer               | null                       | 1680248702 |   1680250125 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3183 | OCA\Talk\BackgroundJob\CheckMatterbridges                       | null                       | 1680249603 |   1680250125 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3184 | OCA\Mail\BackgroundJob\SyncJob                                  | {"accountId":1}            | 1680248702 |   1680250125 |           0 |                  4 | 562423afc287d53e27bc1d34fbc41ce5 |              1 |
|  3185 | OCA\Mail\BackgroundJob\TrainImportanceClassifierJob             | {"accountId":1}            | 1680201902 |   1680250125 |           0 |                  0 | 562423afc287d53e27bc1d34fbc41ce5 |              0 |
|  3232 | OCA\Jira\BackgroundJob\CheckOpenTickets                         | null                       | 1636898401 |   1680250502 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3233 | OCA\Zammad\BackgroundJob\CheckOpenTickets                       | null                       | 1680250126 |   1680250126 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3454 | OC\Security\Bruteforce\CleanupJob                               | null                       | 1680166502 |   1680250126 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  4595 | OCA\Polls\Cron\JanitorCron                                      | null                       | 1680166502 |   1680250126 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  5538 | OCA\Mail\BackgroundJob\CleanupJob                               | null                       | 1680166502 |   1680250126 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  6539 | OCA\DAV\BackgroundJob\CalendarRetentionJob                      | null                       | 1680249603 |   1680250127 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  6540 | OCA\ServerInfo\Jobs\UpdateStorageStats                          | null                       | 1680248706 |   1680250127 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  6544 | OCA\Circles\Cron\Maintenance                                    | null                       | 1680250127 |   1680250127 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  6547 | OCA\Spacedeck\BackgroundJob\CleanupSpacedeck                    | null                       | 1649454302 |   1680250502 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  6566 | OCA\Appointments\Cron\Reminders                                 | null                       | 1680250127 |   1680250127 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  6569 | OCA\Calendar\BackgroundJob\CleanUpOutdatedBookingsJob           | null                       | 1680228012 |   1680250127 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  6571 | OCA\Polls\Cron\AutoReminderCron                                 | null                       | 1680250127 |   1680250127 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  7487 | OCA\Notifications\BackgroundJob\GenerateUserSettings            | null                       | 1680199205 |   1680250127 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  7488 | OCA\Notifications\BackgroundJob\SendNotificationMails           | null                       | 1680250127 |   1680250127 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  7491 | OCA\Richdocuments\Backgroundjobs\Cleanup                        | null                       | 1680248706 |   1680250127 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  8926 | OCA\Mail\BackgroundJob\OutboxWorkerJob                          | null                       | 1680250128 |   1680250127 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10647 | OC\Authentication\Token\TokenCleanupJob                         | null                       | 1680201902 |   1680250128 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
| 10830 | OCA\Passwords\Cron\CheckPasswordsJob                            | null                       | 1680171602 |   1680250128 |           0 |                  2 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10831 | OCA\Passwords\Cron\SynchronizeShares                            | null                       | 1680250129 |   1680250129 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10832 | OCA\Passwords\Cron\SendServerSurvey                             | null                       | 1680053712 |   1680250130 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10833 | OCA\Passwords\Cron\CleanUpEntities                              | null                       | 1680250130 |   1680250130 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10834 | OCA\Passwords\Cron\BackupJob                                    | null                       | 1680171604 |   1680250130 |           0 |                  7 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10929 | OCA\Files\BackgroundJob\DeleteExpiredOpenLocalEditor            | null                       | 1680226513 |   1680250130 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
| 10934 | OCA\Mail\BackgroundJob\PreviewEnhancementProcessingJob          | {"accountId":1}            | 1680248706 |   1680250131 |           0 |                  0 | 562423afc287d53e27bc1d34fbc41ce5 |              1 |
| 10935 | OCA\Mail\BackgroundJob\IMipMessageJob                           | null                       | 1680248706 |   1680250131 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10937 | OCA\Talk\BackgroundJob\ExpireChatMessages                       | null                       | 1680250131 |   1680250131 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 52873 | OCA\Recognize\BackgroundJobs\ClassifyImagenetJob                | {"storageId":2,"rootId":6} | 1680250131 |   1680250131 |  1680250131 |                504 | 4e99bf66762c11ba3abaf152582c224b |              0 |
+-------+-----------------------------------------------------------------+----------------------------+------------+--------------+-------------+--------------------+----------------------------------+----------------+
66 rows in set (0.187 sec)
```

I don't know if it helps, though.

If not, I'll try to continue monitoring the system to see if multiple node processes appear again.

However, currently Face Recognition does not seem to work at all (Object Recognition) seems to be running:

![image](https://user-images.githubusercontent.com/26199042/229065591-8d440967-50f8-44d4-acba-aec14ce4e3ae.png)

Also Audio and Video Recognition do not seem to be running ATM (despite being enabled).

Maybe this "helps" to avoid multiple processes?

Not sure what to look for in the logs regarding this, though.

**Comment by gohrner** [comment]
Please find a current dump of the jobs table - unfortunately after killing the node processes, because it was not possible to run the maridb command line client while they were still stuck in memory:

```plain
MariaDB [nextcloud_db]> SELECT * FROM oc_jobs;
+-------+-----------------------------------------------------------------+-------------------------------------------------------------------------------------------+------------+--------------+-------------+--------------------+----------------------------------+----------------+
| id    | class                                                           | argument                                                                                  | last_run   | last_checked | reserved_at | execution_duration | argument_hash                    | time_sensitive |
+-------+-----------------------------------------------------------------+-------------------------------------------------------------------------------------------+------------+--------------+-------------+--------------------+----------------------------------+----------------+
|     1 | OCA\Files_Versions\BackgroundJob\ExpireVersions                 | null                                                                                      | 1684220290 |   1684221005 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     2 | OCA\Activity\BackgroundJob\EmailNotification                    | null                                                                                      | 1684221005 |   1684221005 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     3 | OCA\Activity\BackgroundJob\ExpireActivities                     | null                                                                                      | 1684185895 |   1684221005 |           0 |              24890 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|     4 | OCA\Files_Sharing\DeleteOrphanedSharesJob                       | null                                                                                      | 1684220328 |   1684221005 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     5 | OCA\Files_Sharing\ExpireSharesJob                               | null                                                                                      | 1684186054 |   1684221005 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|     6 | OCA\Files_Sharing\BackgroundJob\FederatedSharesDiscoverJob      | null                                                                                      | 1684168801 |   1684221005 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     7 | OCA\Support\BackgroundJobs\CheckSubscription                    | null                                                                                      | 1684221005 |   1684221005 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     8 | OCA\UpdateNotification\Notification\BackgroundJob               | null                                                                                      | 1684168801 |   1684221005 |           0 |                153 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|     9 | OCA\Federation\SyncJob                                          | null                                                                                      | 1684196642 |   1684221005 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    10 | OCA\NextcloudAnnouncements\Cron\Crawler                         | null                                                                                      | 1684153502 |   1684221005 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    12 | OCA\DAV\BackgroundJob\CleanupDirectLinksJob                     | null                                                                                      | 1684191578 |   1684221005 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|    13 | OCA\DAV\BackgroundJob\UpdateCalendarResourcesRoomsBackgroundJob | null                                                                                      | 1684218166 |   1684221005 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    14 | OCA\DAV\BackgroundJob\CleanupInvitationTokenJob                 | null                                                                                      | 1684186074 |   1684221005 |           0 |                  4 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|    15 | OCA\Files\BackgroundJob\ScanFiles                               | null                                                                                      | 1684221005 |   1684221005 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    16 | OCA\Files\BackgroundJob\DeleteOrphanedItems                     | null                                                                                      | 1684218166 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    17 | OCA\Files\BackgroundJob\CleanupFileLocks                        | null                                                                                      | 1684220701 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    18 | OCA\Files_Trashbin\BackgroundJob\ExpireTrash                    | null                                                                                      | 1684220292 |   1684221001 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    19 | OC\Authentication\Token\DefaultTokenCleanupJob                  | null                                                                                      | 1665093689 |   1684263353 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    20 | OC\Log\Rotate                                                   | null                                                                                      | 1684221001 |   1684221001 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    21 | OC\Preview\BackgroundCleanupJob                                 | null                                                                                      | 1684221001 |   1684221001 |           0 |                  2 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    75 | OCA\UpdateNotification\ResetTokenBackgroundJob                  | null                                                                                      | 1684221003 |   1684221003 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    81 | OCA\Richdocuments\Backgroundjobs\ObtainCapabilities             | null                                                                                      | 1684221003 |   1684221003 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|    87 | OC\Core\BackgroundJobs\CleanupLoginFlowV2                       | null                                                                                      | 1684217552 |   1684221003 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|   105 | OCA\DAV\BackgroundJob\EventReminderJob                          | null                                                                                      | 1684221004 |   1684221003 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|   106 | OCA\Text\Cron\Cleanup                                           | null                                                                                      | 1684221004 |   1684221004 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|   141 | OCA\Polls\Cron\NotificationCron                                 | null                                                                                      | 1684221005 |   1684221005 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|   234 | OCA\Files\BackgroundJob\CleanupDirectEditingTokens              | null                                                                                      | 1684220302 |   1684221005 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|   236 | OCA\WorkflowEngine\BackgroundJobs\Rotate                        | null                                                                                      | 1684212572 |   1684221005 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  1761 | OCA\ContactsInteraction\BackgroundJob\CleanupJob                | null                                                                                      | 1684191082 |   1684221006 |           0 |                 18 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  3070 | OCA\Contacts\Cron\SocialUpdateRegistration                      | null                                                                                      | 1684174202 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  3083 | OCA\Talk\BackgroundJob\ExpireSignalingMessage                   | null                                                                                      | 1684221006 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3084 | OCA\Talk\BackgroundJob\RemoveEmptyRooms                         | null                                                                                      | 1684221006 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  3085 | OCA\Talk\BackgroundJob\ResetAssignedSignalingServer             | null                                                                                      | 1684221006 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3161 | OCA\Activity\BackgroundJob\DigestMail                           | null                                                                                      | 1684217751 |   1684221006 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3163 | OCA\UserStatus\BackgroundJob\ClearOldStatusesBackgroundJob      | null                                                                                      | 1684221006 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3182 | OCA\Talk\BackgroundJob\CheckHostedSignalingServer               | null                                                                                      | 1684217753 |   1684221006 |           0 |                  5 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3183 | OCA\Talk\BackgroundJob\CheckMatterbridges                       | null                                                                                      | 1684220311 |   1684221006 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3184 | OCA\Mail\BackgroundJob\SyncJob                                  | {"accountId":1}                                                                           | 1684217811 |   1684221006 |           0 |                 93 | 562423afc287d53e27bc1d34fbc41ce5 |              1 |
|  3185 | OCA\Mail\BackgroundJob\TrainImportanceClassifierJob             | {"accountId":1}                                                                           | 1684191434 |   1684221006 |           0 |                  2 | 562423afc287d53e27bc1d34fbc41ce5 |              0 |
|  3232 | OCA\Jira\BackgroundJob\CheckOpenTickets                         | null                                                                                      | 1636898401 |   1684253796 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3233 | OCA\Zammad\BackgroundJob\CheckOpenTickets                       | null                                                                                      | 1684220316 |   1684221006 |           0 |                  4 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  3454 | OC\Security\Bruteforce\CleanupJob                               | null                                                                                      | 1684150801 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  4595 | OCA\Polls\Cron\JanitorCron                                      | null                                                                                      | 1684150801 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  5538 | OCA\Mail\BackgroundJob\CleanupJob                               | null                                                                                      | 1684151702 |   1684221006 |           0 |                  3 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  6539 | OCA\DAV\BackgroundJob\CalendarRetentionJob                      | null                                                                                      | 1684206958 |   1684221006 |           0 |                692 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  6540 | OCA\ServerInfo\Jobs\UpdateStorageStats                          | null                                                                                      | 1684212762 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  6544 | OCA\Circles\Cron\Maintenance                                    | null                                                                                      | 1684221006 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  6547 | OCA\Spacedeck\BackgroundJob\CleanupSpacedeck                    | null                                                                                      | 1649454302 |   1684253797 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  6566 | OCA\Appointments\Cron\Reminders                                 | null                                                                                      | 1684220321 |   1684221006 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  6569 | OCA\Calendar\BackgroundJob\CleanUpOutdatedBookingsJob           | null                                                                                      | 1684217961 |   1684221006 |           0 |                  4 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
|  6571 | OCA\Polls\Cron\AutoReminderCron                                 | null                                                                                      | 1684221006 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  7487 | OCA\Notifications\BackgroundJob\GenerateUserSettings            | null                                                                                      | 1684185573 |   1684221006 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  7488 | OCA\Notifications\BackgroundJob\SendNotificationMails           | null                                                                                      | 1684221006 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  7491 | OCA\Richdocuments\Backgroundjobs\Cleanup                        | null                                                                                      | 1684218012 |   1684221006 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
|  8926 | OCA\Mail\BackgroundJob\OutboxWorkerJob                          | null                                                                                      | 1684221006 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10647 | OC\Authentication\Token\TokenCleanupJob                         | null                                                                                      | 1684191489 |   1684221006 |           0 |                 18 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
| 10830 | OCA\Passwords\Cron\CheckPasswordsJob                            | null                                                                                      | 1684156802 |   1684221006 |           0 |                  1 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10831 | OCA\Passwords\Cron\SynchronizeShares                            | null                                                                                      | 1684221006 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10832 | OCA\Passwords\Cron\SendServerSurvey                             | null                                                                                      | 1684207108 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10833 | OCA\Passwords\Cron\CleanUpEntities                              | null                                                                                      | 1684221006 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10834 | OCA\Passwords\Cron\BackupJob                                    | null                                                                                      | 1684156803 |   1684221006 |           0 |                  2 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10929 | OCA\Files\BackgroundJob\DeleteExpiredOpenLocalEditor            | null                                                                                      | 1684220704 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              0 |
| 10934 | OCA\Mail\BackgroundJob\PreviewEnhancementProcessingJob          | {"accountId":1}                                                                           | 1684218120 |   1684221006 |           0 |                 15 | 562423afc287d53e27bc1d34fbc41ce5 |              1 |
| 10935 | OCA\Mail\BackgroundJob\IMipMessageJob                           | null                                                                                      | 1684218122 |   1684221006 |           0 |                 41 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 10937 | OCA\Talk\BackgroundJob\ExpireChatMessages                       | null                                                                                      | 1684221006 |   1684221006 |           0 |                  0 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 53681 | OCA\Memories\Cron\IndexJob                                      | null                                                                                      | 1684220329 |   1684221006 |           0 |                264 | 37a6259cc0c1dae299a7866489dff0bd |              1 |
| 55379 | OCA\Recognize\BackgroundJobs\ClassifyLandmarksJob               | {"storageId":2,"rootId":6}                                                                | 1684221004 |   1684221004 |           0 |               2692 | 4e99bf66762c11ba3abaf152582c224b |              0 |
| 55392 | OCA\Recognize\BackgroundJobs\StorageCrawlJob                    | {"storage_id":2,"root_id":6,"override_root":7,"last_file_id":53251,"models":["imagenet"]} |          0 |   1684221006 |           0 |                  0 | fc0a21538ca55bb0d3439cfb7fb97bce |              1 |
+-------+-----------------------------------------------------------------+-------------------------------------------------------------------------------------------+------------+--------------+-------------+--------------------+----------------------------------+----------------+
68 rows in set (0.000 sec)
```

**Comment by gohrner** [comment]
@marcelklehr: As I cannot intentionally trigger it, and once it happens it's close to impossible to do anything with this server, it's a bit difficult to provide this feedback.

As soon as it happens again, I'll try to remember to perform a detailed `pstree` before killing the processes.

**Comment by DavidMndz** [comment]
Just a dumb question, might be it related to the PHP version? It is not mentioned the version used by Debian 11 uses 7.4, while the Recognize app seems to require at least PHP 8.0.

**Comment by gohrner** [comment]
@marcelklehr: Today, I noticed it again.

The answer looks like "different", here's `pstree -p -g`:

```plain
             ├─cron(1012,1012)─┬─cron(113906,1012)───sh(113911,113911)───flock(113920,113911)───python3(113921,113911)───{python3}(113979,1+
             │                 ├─cron(115181,1012)───sh(115187,115187)───php(115188,115187)───node(115453,115187)─┬─{node}(115460,115187)
             │                 │                                                                                  ├─{node}(115463,115187)
             │                 │                                                                                  ├─{node}(115464,115187)
             │                 │                                                                                  ├─{node}(115467,115187)
             │                 │                                                                                  ├─{node}(115469,115187)
             │                 │                                                                                  ├─{node}(115481,115187)
             │                 │                                                                                  ├─{node}(115534,115187)
             │                 │                                                                                  ├─{node}(115536,115187)
             │                 │                                                                                  ├─{node}(115541,115187)
             │                 │                                                                                  ├─{node}(115544,115187)
             │                 │                                                                                  ├─{node}(115573,115187)
             │                 │                                                                                  ├─{node}(115574,115187)
             │                 │                                                                                  ├─{node}(115575,115187)
             │                 │                                                                                  └─{node}(115576,115187)
             │                 ├─cron(115359,1012)───sh(115361,115361)───php(115363,115361)───node(115456,115361)─┬─{node}(115461,115361)
             │                 │                                                                                  ├─{node}(115471,115361)
             │                 │                                                                                  ├─{node}(115473,115361)
             │                 │                                                                                  ├─{node}(115474,115361)
             │                 │                                                                                  ├─{node}(115476,115361)
             │                 │                                                                                  ├─{node}(115480,115361)
             │                 │                                                                                  ├─{node}(115538,115361)
             │                 │                                                                                  ├─{node}(115539,115361)
             │                 │                                                                                  ├─{node}(115546,115361)
             │                 │                                                                                  ├─{node}(115548,115361)
             │                 │                                                                                  ├─{node}(115569,115361)
             │                 │                                                                                  ├─{node}(115570,115361)
             │                 │                                                                                  ├─{node}(115571,115361)
             │                 │                                                                                  └─{node}(115572,115361)
             │                 ├─cron(115403,1012)───sh(115405,115405)───php(115406,115405)───node(115450,115405)─┬─{node}(115462,115405)
             │                 │                                                                                  ├─{node}(115472,115405)
             │                 │                                                                                  ├─{node}(115475,115405)
             │                 │                                                                                  ├─{node}(115477,115405)
             │                 │                                                                                  ├─{node}(115478,115405)
             │                 │                                                                                  ├─{node}(115482,115405)
             │                 │                                                                                  ├─{node}(115533,115405)
             │                 │                                                                                  ├─{node}(115535,115405)
             │                 │                                                                                  ├─{node}(115542,115405)
             │                 │                                                                                  ├─{node}(115543,115405)
             │                 │                                                                                  ├─{node}(115565,115405)
             │                 │                                                                                  ├─{node}(115566,115405)
             │                 │                                                                                  ├─{node}(115567,115405)
             │                 │                                                                                  └─{node}(115568,115405)
             │                 ├─cron(115585,1012)───sh(115586,115586)───php(115587,115586)───node(116066,115586)─┬─{node}(116069,115586)
             │                 │                                                                                  ├─{node}(116070,115586)
             │                 │                                                                                  ├─{node}(116071,115586)
             │                 │                                                                                  ├─{node}(116072,115586)
             │                 │                                                                                  ├─{node}(116073,115586)
             │                 │                                                                                  ├─{node}(116074,115586)
             │                 │                                                                                  ├─{node}(116108,115586)
             │                 │                                                                                  ├─{node}(116109,115586)
             │                 │                                                                                  ├─{node}(116114,115586)
             │                 │                                                                                  ├─{node}(116115,115586)
             │                 │                                                                                  ├─{node}(116134,115586)
             │                 │                                                                                  ├─{node}(116135,115586)
             │                 │                                                                                  ├─{node}(116136,115586)
             │                 │                                                                                  └─{node}(116137,115586)
             │                 ├─cron(115658,1012)───sh(115684,115684)───php(115685,115684)───node(116032,115684)─┬─{node}(116048,115684)
             │                 │                                                                                  ├─{node}(116049,115684)
             │                 │                                                                                  ├─{node}(116050,115684)
             │                 │                                                                                  ├─{node}(116051,115684)
             │                 │                                                                                  ├─{node}(116052,115684)
             │                 │                                                                                  ├─{node}(116062,115684)
             │                 │                                                                                  ├─{node}(116105,115684)
             │                 │                                                                                  ├─{node}(116107,115684)
             │                 │                                                                                  ├─{node}(116111,115684)
             │                 │                                                                                  ├─{node}(116113,115684)
             │                 │                                                                                  ├─{node}(116138,115684)
             │                 │                                                                                  ├─{node}(116139,115684)
             │                 │                                                                                  ├─{node}(116140,115684)
             │                 │                                                                                  └─{node}(116141,115684)
             │                 ├─cron(115862,1012)───sh(115876,115876)───php(115877,115876)───node(116035,115876)─┬─{node}(116054,115876)
             │                 │                                                                                  ├─{node}(116055,115876)
             │                 │                                                                                  ├─{node}(116056,115876)
             │                 │                                                                                  ├─{node}(116057,115876)
             │                 │                                                                                  ├─{node}(116058,115876)
             │                 │                                                                                  ├─{node}(116063,115876)
             │                 │                                                                                  ├─{node}(116104,115876)
             │                 │                                                                                  ├─{node}(116106,115876)
             │                 │                                                                                  ├─{node}(116110,115876)
             │                 │                                                                                  ├─{node}(116112,115876)
             │                 │                                                                                  ├─{node}(116142,115876)
             │                 │                                                                                  ├─{node}(116143,115876)
             │                 │                                                                                  ├─{node}(116144,115876)
             │                 │                                                                                  └─{node}(116145,115876)
             │                 ├─cron(116019,1012)───sh(116021,116021)───php(116027,116021)───node(116092,116021)─┬─{node}(116095,116021)
             │                 │                                                                                  ├─{node}(116096,116021)
             │                 │                                                                                  ├─{node}(116097,116021)
             │                 │                                                                                  ├─{node}(116098,116021)
             │                 │                                                                                  ├─{node}(116099,116021)
             │                 │                                                                                  ├─{node}(116100,116021)
             │                 │                                                                                  ├─{node}(116117,116021)
             │                 │                                                                                  ├─{node}(116118,116021)
             │                 │                                                                                  ├─{node}(116119,116021)
             │                 │                                                                                  ├─{node}(116120,116021)
             │                 │                                                                                  ├─{node}(116130,116021)
             │                 │                                                                                  ├─{node}(116131,116021)
             │                 │                                                                                  ├─{node}(116132,116021)
             │                 │                                                                                  └─{node}(116133,116021)
             │                 ├─cron(116124,1012)───sh(116125,116125)───flock(116126,116125)───python3(116127,116125)───{python3}(116228,1+
             │                 ├─cron(116219,1012)───sh(116220,116220)───php(116223,116220)───node(116499,116220)─┬─{node}(116512,116220)
             │                 │                                                                                  ├─{node}(116513,116220)
             │                 │                                                                                  ├─{node}(116514,116220)
             │                 │                                                                                  ├─{node}(116515,116220)
             │                 │                                                                                  ├─{node}(116516,116220)
             │                 │                                                                                  └─{node}(116523,116220)
             │                 ├─cron(116368,1012)───sh(116372,116372)───php(116377,116372)───{php}(116404,116372)
             │                 ├─cron(116369,1012)───sh(116370,116370)───php(116378,116370)───node(116534,116370)─┬─{node}(116537,116370)
             │                 │                                                                                  ├─{node}(116538,116370)
             │                 │                                                                                  ├─{node}(116539,116370)
             │                 │                                                                                  ├─{node}(116540,116370)
             │                 │                                                                                  ├─{node}(116541,116370)
             │                 │                                                                                  └─{node}(116542,116370)
             │                 ├─cron(116435,1012)───sh(116438,116438)───flock(116440,116438)───python3(116444,116438)───{python3}(116553,1+
             │                 └─cron(116560,1012)───sh(116568,116568)───php(116570,116568)───{php}(116579,116568)
```

**Comment by gohrner** [comment]
@marcelklehr: I'm confused about the automated message today. After the issue finally occurred again, I provided all information I knew was requested in my last posting two weeks ago.

What else are you currently waiting for, or why was this reminder added? Did I somehow misconfigure the issue?

**Comment by marcelklehr** [comment]
> I was actually waiting for your feedback since these two weeks.

Yes, I'm sorry. The bot sometimes gets overzealous :( I had the issue on "waiting for more information" but didn't have time to remove the label after you commented so the bot thought nobody had replied... All good :hearts:

**Comment by marcelklehr** [comment]
> Before, it had 0 files in the queue for face recognition, now it's more than 1000 files in the queue...

When enabling various classifiers, background jobs are scheduled automatically, yes.

> The answer looks like "different cron processes", here's pstree -p -g:

Thank you for digging into this! This is what I suspected. The cron jobs are using up 6 cores each, it seems, but they should only run for 5mins max and then get killed. It may be that Nextcloud's cron.php just picks up the next waiting recognize jobs and ploughs on. Then I think the only solution is to use advanced scheduling mechanisms in Nextcloud 27, which allow checking if other jobs are currently running and avoiding running the same job multiple times in parallel.

Thanks again for your efforts!

**Comment by Forza-tng** [comment]
> Then I think the only solution is to use advanced scheduling mechanisms in Nextcloud 27, which allow checking if other jobs are currently running and avoiding running the same job multiple times in parallel.


Can't we use a lock, either as a file or a db/redis entry? If the lock is active, the recognise job could simply be a noop/skip? Might be useful to add a a few second delay so the cron job doesn't spawn too fast?

**Comment by marcelklehr** [comment]
> Can't we use a lock, either as a file or a db/redis entry? If the lock is active, the recognise job could simply be a noop/skip? Might be useful to add a a few second delay so the cron job doesn't spawn too fast?

Yes, that would also be an option, good point :+1:

**Comment by marcelklehr** [comment]
This should be finally be fixed with the release of v27.1.4 next week :tada:
